pulsebot/bugzilla.py

```diff
@@ -17,85 +17,50 @@ class BugzillaError(Exception):
 class Bugzilla(object):
     def __init__(self, server, api_key):
         self._server = server.rstrip("/")
-        self._session = requests.Session()
-        self._session.headers.update({"User-Agent": "pulsebot"})
-        self._session.headers.update({"X-Bugzilla-API-Key": api_key})
-
-    def get_fields(self, bug, fields):
-        bug_url = "%s/rest/bug/%d?include_fields=%s" % (
-            self._server,
-            bug,
-            ",".join(fields),
-        )
-        logger.info(f"get_fields: {bug_url}")
+        self._headers = {
+            "User-Agent": "pulsebot",
+            "X-Bugzilla-API-Key": api_key,
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+
+    def _call(self, method, path, data={}):
+        if not method:
+            method = "GET"
+        bug_url = f"{self._server}/{path}"
+        logger.info(f"{method}: {bug_url}")
         try:
-            r = self._session.get(bug_url)
+            r = requests.request(method, bug_url, headers=self._headers, data=data)
```

Comment:
`data` is actually unused for `GET` calls, which makes this a little misleading. I think that it would still be cleaner to use `requests.get`, `requests.post`, etc... here, and we could proxy them through the specified `method` if needed. But, if you want to use `requests.request`, we could just check the `method` and then not pass `data` if it is not needed.